File: Sunrin-ICPC-2022/Pre-C/PC.py
T = int(input())

for tc in range(T):
    n = int(input())
    a = list(map(int, input().split()))
    flag = True
    for i in range(n):
        a[i] = a[i] if a[i] < n-a[i]+1 else n-a[i]+1
        if i > 0 and a[i-1] > a[i]: a[i] = n-a[i]+1
        if i > 0 and a[i-1] > a[i]: flag = False
    print('YES' if flag else 'NO')

File: Sunrin-ICPC-2022/Pre-C/PC.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const T = parseInt(input[0]);
for(let i=1; i<=T; i++){
    const n = parseInt(input[i*2-1]);
    const a = input[i*2].split(' ').map(s => parseInt(s));
    if(a.length != n) break;
    let flag = true;
    for(let j=0; j<n; j++){
        a[j] = Math.min(a[j], n - a[j] + 1);
        if(j > 0 && a[j-1] > a[j]) a[j] = n - a[j] + 1;
        if(j > 0 && a[j-1] > a[j]) flag = false;
    }
    if(flag) console.log("YES");
    else console.log("NO");
}